pennylane_lightning_gpu/src/tests/Test_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -930,3 +930,67 @@ TEMPLATE_TEST_CASE("StateVectorCudaManaged::Hamiltonian_expval_cuSparse",
         CHECK(expected == Approx(results).epsilon(1e-7));
     }
 }
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::SetStates",
+                   "[StateVectorCudaManaged_Nonparam]", float, double) {
+    using cp_t = std::complex<TestType>;
+    const std::size_t num_qubits = 3;
+
+    SECTION("SetStates") {
+        std::vector<cp_t> init_state{{0.0, 0.0}, {0.0, 0.1}, {0.1, 0.1},
+                                     {0.1, 0.2}, {0.2, 0.2}, {0.3, 0.3},
+                                     {0.3, 0.4}, {0.4, 0.5}};
+        std::vector<cp_t> expected_state{{2.0, 0.0},  {0.0, 0.1}, {4.0, 0.0},
+                                         {0.1, 0.2},  {8.0, 0.0}, {0.3, 0.3},
+                                         {12.0, 0.0}, {0.4, 0.5}};
```

Comment:
We should probably opt for a normalised state result, even though I know this will functionally fine. Best to keep it physically correct too.

Reply:
np!